pkg/controller/controller.go

```diff
@@ -58,6 +58,10 @@ type Options struct {
 	// RecoverPanic indicates whether the panic caused by reconcile should be recovered.
 	// Defaults to the Controller.RecoverPanic setting from the Manager if unset.
 	RecoverPanic *bool
+
+	// ControllerCtx is the optional context for only this Controller. If it is set and been canceled,
+	// this controller and its watches will be stopped dynamically.
+	ControllerCtx context.Context
```

Comment:
Why not add a `Stop` method instead?

Reply:
Yeah, as I asked in https://github.com/kubernetes-sigs/controller-runtime/pull/2099#issuecomment-1350370602 .
Now I have changed the `ControllerCtx` to a `Stop` method.